ui/src/clientLibraries/components/ClientRubyOverlay.tsx

```diff
@@ -0,0 +1,125 @@
+
+// Components
+import ClientLibraryOverlay from 'src/clientLibraries/components/ClientLibraryOverlay'
+import TemplatedCodeSnippet from 'src/shared/components/TemplatedCodeSnippet'
+
+// Constants
+import {clientRubyLibrary} from 'src/clientLibraries/constants'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+
+// Types
+import {AppState} from 'src/types'
+
+interface StateProps {
+  org: string
+}
+
+type Props = StateProps
+
+const ClientRubyOverlay: FunctionComponent<Props> = props => {
+  const {
+    name,
+    url,
+    initializeGemCodeSnippet,
+    initializeClientCodeSnippet,
+    writingDataLineProtocolCodeSnippet,
+    writingDataPointCodeSnippet,
+    writingDataHashCodeSnippet,
+    writingDataBatchCodeSnippet,
+  } = clientRubyLibrary
+  const {org} = props
+  const server = window.location.origin
+
+  return (
+    <ClientLibraryOverlay title={`${name} Client Library`}>
+      <p>
+        For more detailed and up to date information check out the{' '}
+        <a href={url} target="_blank">
+          GitHub Repository
+        </a>
+      </p>
+      <h5>Install the Gem</h5>
+      <TemplatedCodeSnippet template={initializeGemCodeSnippet} label="Code" />
+      <h5>Initialize the Client</h5>
+      <TemplatedCodeSnippet
+        template={initializeClientCodeSnippet}
+        label="Ruby Code"
+        defaults={{
+          server: 'serverUrl',
+          token: 'token',
+        }}
+        values={{
+          server,
+        }}
+      />
+      <h5>Write Data</h5>
+      <p>Option 1: Use InfluxDB Line Protocol to write data</p>
+      <TemplatedCodeSnippet
+        template={writingDataLineProtocolCodeSnippet}
+        label="Ruby Code"
+        defaults={{
+          bucket: 'bucketID',
+          org: 'orgID',
+        }}
+        values={{
+          org,
+        }}
+      />
+      <p>Option 2: Use a Data Point to write data</p>
+      <TemplatedCodeSnippet
+        template={writingDataPointCodeSnippet}
+        label="Ruby Code"
+        defaults={{
+          bucket: 'bucketID',
+          org: 'orgID',
+        }}
+        values={{
+          org,
+        }}
+      />
+      <p>Option 3: Use a Hash to write data</p>
+      <TemplatedCodeSnippet
+        template={writingDataHashCodeSnippet}
+        label="Ruby Code"
+        defaults={{
+          bucket: 'bucketID',
+          org: 'orgID',
+        }}
+        values={{
+          org,
+        }}
+      />
+      <p>Option 4: Use a Batch Sequence to write data</p>
+      <TemplatedCodeSnippet
+        template={writingDataBatchCodeSnippet}
+        label="Ruby Code"
+        defaults={{
+          bucket: 'bucketID',
+          org: 'orgID',
+        }}
+        values={{
+          org,
+        }}
+      />
+    </ClientLibraryOverlay>
+  )
+}
+
+const mstp = (state: AppState): StateProps => {
+  const {id} = getOrg(state)
+
+  return {
+    org: id,
+  }
+}
+
+export {ClientRubyOverlay}
+export default connect<StateProps, {}, Props>(
```

Comment:
i think the last parameter on this is supposed to be `{}` as it represents the all the props being passed in that aren't part of state and dispatch props, but if the typescript checks are happy, i'm happy